src/common_types.hpp

```diff
@@ -53,6 +53,18 @@ DEFINE_LEXICOGRAPHIC_ORDERING(cell_member_type,(a.gid,a.index),(b.gid,b.index))
 
 using time_type = float;
 
+// Enumeration used to indentify the cell type/kind, used by the model to
+// group equal kinds in the same cell group.
+
+enum cell_kind {
+    multicompartment,
+    poisson,
+    from_file,
```

Comment:
I would leave it out until you are ready to include it in the PR.
Indeed, if we are going to make a minimal PR, then there should just be the one `cell_kind`, i.e. `multicompartment`.

Regarding the semantics, @halfflat has a point. The multicompartment solution discretization is on the level of the `fvm_multicell`.

For the "poisson" cell types, it depends how specific we make the cell groups. We could have a cell group just for a generalised Poisson cell type, and cell group specializations for  other spike train generating cell types (e.g. cell groups that output pre recorded cells from file).